qi-lib/flow/core/compiler/deforest/cps.rkt

```diff
@@ -198,29 +198,30 @@
        ;; A static runtime contract is placed at the beginning of the
        ;; fused sequence. And runtime checks for consumers are in
        ;; their respective implementation procedure.
-       #`(esc
-          (#,((attribute p.curry) ctx (attribute p.name))
-           (contract p.contract
-                     (p.prepare
-                      (lambda (state)
-                        (define cstate (inline-consing state t.state ...))
-                        cstate)
-                      (#,@#'c.end
-                       (inline-compose1 [t.next t.f
-                                                '#,(prettify-flow-syntax ctx)
-                                                '#,(build-source-location-vector
-                                                    (syntax-srcloc ctx))
-                                                ] ...
-                                        p.next
-                                        )
+       (with-syntax (((rt ...) (reverse (attribute t.state))))
+         #`(esc
+            (#,((attribute p.curry) ctx (attribute p.name))
+             (contract p.contract
+                       (p.prepare
+                        (lambda (state)
+                          (define cstate (inline-consing state rt ...))
+                          cstate)
```

Comment:
Is this a fix for the inline-consing from 1a79dc0 (Implement proper state
cons-ing for take., 2024-05-31)? I'd rather see it as part of that commit, then.

Also, is `#:with` viable here to avoid extra nesting depth? (I know it has some
semantic differences from `with-syntax` for raw datums, but I don't think that
has an impact on `t.state` + `inline-consing`?)

Reply:
Addressed during rebases. The suggestion of #:with is something to think about after resolving the datum/literal matching - which should be done after merging the branch with deforestation support in the core language.